src/main/java/com/truelayer/java/TrueLayerClientBuilder.java

```diff
@@ -134,13 +188,29 @@ public TrueLayerClientBuilder withCredentialsCaching(ICredentialsCache credentia
      * @see TrueLayerClient
      */
     public TrueLayerClient build() {
-        if (ObjectUtils.isEmpty(clientCredentials)) {
+        if (isEmpty(clientCredentials)) {
             throw new TrueLayerException("client credentials must be set");
         }
 
         VersionInfo versionInfo = new VersionInfoLoader().load();
         OkHttpClient.Builder clientBuilder = new OkHttpClient.Builder();
 
+        if (isNotEmpty(timeout)) {
```

Comment:
it's not straightforward to unit test this and below as per the current design. Whilst I created an integration test to verify timeout, I did not introduce tests for below 2 options. If we really want to test what this builder is doing we should probably redesign a bit this class. That being said, I'm happy with the current shape of it as of now.